scripts/validation/validate-examples.js

```diff
@@ -0,0 +1,35 @@
+const { execSync } = require('child_process');
+const glob = require('glob');
+
+// Use glob to find all YAML files in the examples directory
+const files = glob.sync('./examples/**/*.{yml,yaml}');
+
+let filesCount = 0;
+let errorFilesCount = 0;
+let filesWithErrors = []; // Array to store files that failed validation
+// Validate each file using AsyncAPI CLI
+files.forEach((file) => {
+  filesCount++;
+  try {
+    console.log(`\nValidating: ${file}`);
+    execSync(`npx asyncapi validate ${file}`, { stdio: 'inherit' });
```

Comment:
This is gonna be pretty slow. Can we run it async somehow and compare performance? 🤔

Reply:
> This is gonna be pretty slow. Can we run it async somehow and compare performance? 🤔

Okay, let me try doing it asynchronously...

Reply:
Applied suggestion through baa7c12

Reply:
# Performance Report

## Before
When the script was executed _**synchronously**_:
- [Job link](https://github.com/AnimeshKumar923/asyncapi-spec/actions/runs/8410069994/usage)
- _Screenshot_
![image](https://github.com/asyncapi/spec/assets/99868037/991bfcb3-6759-4591-94d5-3aa72cdb3ae0)

---
## After
When the script was executed _**asynchronously**_:
- [Job link](https://github.com/AnimeshKumar923/asyncapi-spec/actions/runs/8410126555/usage)
- _Screenshot_
![image](https://github.com/asyncapi/spec/assets/99868037/0a1eeac6-9c11-47a3-ba49-bdd29812df25)
---

Conclusion: The execution time here has been **reduced by 1m 10s** with the use of asynchronous way of JavsScript.

Thanks for the suggestion! @smoya :+1:

Reply:
## The execution time for the script itself

### Before
![image](https://github.com/asyncapi/spec/assets/99868037/df9d3358-0202-4639-84a7-27e561c6af65)

### After
![image](https://github.com/asyncapi/spec/assets/99868037/fadf4435-66bc-4ac6-b8f7-586ec000a0c6)

Reply:
> I think we can go further and skip using a JS script here and find a simple online terminal command that do the job.
> 
> I believe by using `find` which also allows excluding paths + piping to `asyncapi` command (CLI) through `xargs` would end up being a really simple oneliner, not complicated and simple to read and maintain.

Let me look into that...

Reply:
@smoya The code has been successfully executed. Please have a look at this [workflow job detail](https://github.com/AnimeshKumar923/asyncapi-spec/actions/runs/8573974272/job/23499821262)

---
### Attaching screenshot for more details:
![image](https://github.com/asyncapi/spec/assets/99868037/921e9815-b155-4359-9e4b-53123134d6ea)

---
### Job run details
![image](https://github.com/asyncapi/spec/assets/99868037/3e6be1a6-cfdf-40c1-9cae-3f22b582f3cd)
_[Link for reference](https://github.com/AnimeshKumar923/asyncapi-spec/actions/runs/8573974272/usage)_